### ISSUE TITLE
When adding a audit log entry for a search that has included resource, add a count of resource that were included.

### ISSUE THREAD

**Issue by d0roppe** [IBM/FHIR#2073]
**Is your feature request related to a problem? Please describe.**
In an audit entry for a search there is a read entry with a content containing the count of matching searches. but nothing mentioned about how many included resource. 
**Describe the solution you'd like**
Add the count of included resources to the audit log entry if possible by resource type. but at least the overall count of included resources. 
**Describe alternatives you've considered**

**Acceptance Criteria**
1.  
   GIVEN [a precondition]
    AND [another precondition]
   WHEN [test step]
    AND [test step]
   THEN [verification step]
    AND [verification step]

**Additional context**

**Comment by lmsurpre** [comment]
I still think it will be more interesting if we can include all the ids for the resources that are returned to the client as part of the search.  This would let us answer questions like "which users accessed version 2 of resource Patient/123 in January of 2021".

**Comment by prb112** [comment]
we'd really have to test this to see the total payload size is under 1M... this is a good idea, just a lot edge cases

**Comment by lmsurpre** [comment]
Yes, I feel pretty good about it for "normal" search (where we have a 1000-resource limit).  Less confident on bulk export...do we emit a separate audit event for each page exported or one big one?

**Comment by prb112** [comment]
Each page.

**Comment by lmsurpre** [comment]
> Each page.

OK, then I feel like it should be ok.  Assuming we used some kind of string entry like "<resourceType>/<id>/_history/<version>, here's my rough back-of-envelope calculations.

Average case:
```
entry wrapper (need to see how this looks in CADF to be sure):  1 byte * 12 chars = 12 bytes
resourceType/:  1 byte * 20 chars = 20 bytes
id/:  1 byte * 50 chars for value = 50 bytes
_history/:  1 byte * 9 chars = 9 bytes
version:  1 byte * 2 chars = 2 bytes
= 93 bytes per entry
2000 entries * 93 = 186,000 bytes (.186 MB)
```

Worst case:
```
entry wrapper (need to see how this looks in CADF to be sure):  1 byte * 12 chars = 12 bytes
resourceType/:  1 byte * 40 chars = 40 bytes
id/:  4 byte * 64 chars for value = 256 bytes
_history/:  1 byte * 9 chars = 9 bytes
version:  1 byte * 64 chars = 64 bytes
= 381 bytes per entry

2000 entries * 381 = 762,000 bytes (.762 MB)
```

**Comment by prb112** [comment]
It's really a different work item beyond what's described here.

**Comment by lmsurpre** [comment]
Yeah, I guess I don't understand why we'd care about the count of resources.

**Comment by prb112** [comment]
This carries forward the RESTAuditLogger's approach to audit. If we want to change, it's a different workitem.

**Comment by lmsurpre** [comment]
OK, I opened https://github.com/IBM/FHIR/issues/2090 for it.
